E2E Activity Feed tests fixed

nigel-wells opened the pull request:
Have updated all SF E2E tests for the new activity feed layout.
All tests are passing for SF & LF (except for the usual 3).
There were two activities that were not tracking the `userRef`. I've added in the `userRef` based on how I saw it being used in other parts of the API - this will need to be reviewed in case there is a more official way of doing it.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/sillsdev/web-languageforge/255)
<!-- Reviewable:end -->

irahopkinson commented:
BTW remember to run the PHP unit tests since you have changed PHP files. They are currently failing probably because of the `IdReference` issue below.

---

Review status: all files reviewed at latest revision, 8 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io/reviews/sillsdev/web-languageforge/255#-:-LCX1cGoA2KXQq5qXWIQ:bf88dpq)*
<!-- Sent from Reviewable.io -->

nigel-wells commented:
I've fixed up the unit tests - completely forgot about those ones. There are 3 failing but I believe they are unrelated:

```
Failed asserting that an object is empty.
 /mnt/c/src/web-languageforge/test/php/library/shared/LanguageDataTest.php:31
 

Failed asserting that an array does not have the key 'gax'.
 /mnt/c/src/web-languageforge/test/php/library/shared/LanguageDataTest.php:43
 

The file "/usr/share/GeoIP/GeoLite2-City.mmdb" does not exist or is not readable.
 /mnt/c/src/web-languageforge/src/Api/Model/Shared/Translate/Dto/TranslateMetricDto.php:102
 /mnt/c/src/web-languageforge/test/php/model/shared/translate/dto/TranslateMetricDtoTest.php:33
```
 
That last error is because the file doesn't exist - can I get a copy of it from someone?

---

Review status: all files reviewed at latest revision, 8 unresolved discussions, some commit checks failed.

---

*[src/Api/Model/Scriptureforge/Sfchecks/Command/TextCommands.php, line 54 at r1](https://reviewable.io/reviews/sillsdev/web-languageforge/255#-LCWtpLa3lQMvCPw0nbp:-LCX3HP224FlP2SHKCy1:b-b4xjv) ([raw file](https://github.com/sillsdev/web-languageforge/blob/078c6266a8acc3bebfc8dbcf3cac0c7bf776297d/src/Api/Model/Scriptureforge/Sfchecks/Command/TextCommands.php#L54)):*
<details><summary><i>Previously, irahopkinson (Ira Hopkinson) wrote…</i></summary><blockquote>

> ```
> 	/**
> 	 * @param string $projectId
> 	 * @param array $object (json encoded)
> 	 * @param $userId
> 	 *
> 	 * @return string Id of text updated/added
> 	 * @throws \Api\Library\Shared\Palaso\Exception\ResourceNotAvailableException
> 	 */
> ```
Your editor is putting actual tab characters in instead of 4 spaces per tab. We don't want tab characters after all the work to get them out.
</blockquote></details>

Done - have updated IDE settings to stop this happening again as well.

---

*[src/Api/Model/Shared/Command/ActivityCommands.php, line 105 at r1](https://reviewable.io/reviews/sillsdev/web-languageforge/255#-LCX-XUu3_mK8Y4RZ0R8:-LCX3TQEDaIW-9tFmzFi:b-896fix) ([raw file](https://github.com/sillsdev/web-languageforge/blob/078c6266a8acc3bebfc8dbcf3cac0c7bf776297d/src/Api/Model/Shared/Command/ActivityCommands.php#L105)):*
<details><summary><i>Previously, irahopkinson (Ira Hopkinson) wrote…</i></summary><blockquote>

> ```
> 	 * @param string $textId
> 	 * @param TextModel $textModel
> 	 * @param string $userId
> 	 *
> 	 * @return string activity id
> 	 * @throws \Exception
> ```
Your editor is putting actual tab characters in instead of 4 spaces per tab. We don't want tab characters after all the work to get them out.
</blockquote></details>

Done.

---

*[src/Api/Model/Shared/Command/ActivityCommands.php, line 116 at r1](https://reviewable.io/reviews/sillsdev/web-languageforge/255#-LCWw4exAej2Qn53p-FD:-LCX3pvzCQQwiIZx6deY:b-896fix) ([raw file](https://github.com/sillsdev/web-languageforge/blob/078c6266a8acc3bebfc8dbcf3cac0c7bf776297d/src/Api/Model/Shared/Command/ActivityCommands.php#L116)):*
<details><summary><i>Previously, irahopkinson (Ira Hopkinson) wrote…</i></summary><blockquote>

> ```
> $activity->userRef
> ```
This is an `IdReference` type so you need to assign it like the others, i.e. `$activity->userRef->id = $userId;`. I had a quick look at the Activity List DTO and it should come back to the front end as a string still.
</blockquote></details>

Done.

---

*[src/Api/Model/Shared/Command/ActivityCommands.php, line 132 at r1](https://reviewable.io/reviews/sillsdev/web-languageforge/255#-LCX-ZQOE9CSvdO8E_VV:-LCX3gjtExR_mz57MdA9:b-896fix) ([raw file](https://github.com/sillsdev/web-languageforge/blob/078c6266a8acc3bebfc8dbcf3cac0c7bf776297d/src/Api/Model/Shared/Command/ActivityCommands.php#L132)):*
<details><summary><i>Previously, irahopkinson (Ira Hopkinson) wrote…</i></summary><blockquote>

> ```
> 	/**
> 	 * @param ProjectModel $projectModel
> 	 * @param string $questionId
> 	 * @param QuestionModel $questionModel
> 	 * @param string $userId
> 	 *
> 	 * @return string activity id
> 	 * @throws \Exception
> 	 */
> ```
Your editor is putting actual tab characters in instead of 4 spaces per tab. We don't want tab characters after all the work to get them out.
</blockquote></details>

Done.

---

*[src/Api/Model/Shared/Command/ActivityCommands.php, line 139 at r1](https://reviewable.io/reviews/sillsdev/web-languageforge/255#-LCWwExuCSAZ955-Whkj:-LCX3vYFAhIRAQrVrr73:b-896fix) ([raw file](https://github.com/sillsdev/web-languageforge/blob/078c6266a8acc3bebfc8dbcf3cac0c7bf776297d/src/Api/Model/Shared/Command/ActivityCommands.php#L139)):*
<details><summary><i>Previously, irahopkinson (Ira Hopkinson) wrote…</i></summary><blockquote>

> ```
> $activity->userRef
> ```
This is an `IdReference` type so you need to assign it like the others, i.e. `$activity->userRef->id = $userId;`
</blockquote></details>

Done.

---

*[test/app/bellows/shared/activity.page.ts, line 3 at r1](https://reviewable.io/reviews/sillsdev/web-languageforge/255#-LCWwvxg6ZefUO8NgyVW:-LCX41HP02lpG9LS38Au:b5dfu6x) ([raw file](https://github.com/sillsdev/web-languageforge/blob/078c6266a8acc3bebfc8dbcf3cac0c7bf776297d/test/app/bellows/shared/activity.page.ts#L3)):*
<details><summary><i>Previously, irahopkinson (Ira Hopkinson) wrote…</i></summary><blockquote>

> ```
> "./utils"
> ```
Style: use single quotes.
</blockquote></details>

Have removed completely as wasn't required any more by the looks

---

*[test/app/bellows/shared/activity.page.ts, line 3 at r1](https://reviewable.io/reviews/sillsdev/web-languageforge/255#-LCWxIyg08CONgfTw53V:-LCX48BB-dyNcKL9a6k2:b-dshica) ([raw file](https://github.com/sillsdev/web-languageforge/blob/078c6266a8acc3bebfc8dbcf3cac0c7bf776297d/test/app/bellows/shared/activity.page.ts#L3)):*
<details><summary><i>Previously, irahopkinson (Ira Hopkinson) wrote…</i></summary><blockquote>

Style: separate external and internal imports by placing a blank line between them.
</blockquote></details>

Removed as noted above

---

*[test/app/scriptureforge/activity/activity.e2e-spec.ts, line 512 at r1](https://reviewable.io/reviews/sillsdev/web-languageforge/255#-LCWzATA8ORFH3LLVCNp:-LCX67zK1xABmJSFs7Xn:b-896fix) ([raw file](https://github.com/sillsdev/web-languageforge/blob/078c6266a8acc3bebfc8dbcf3cac0c7bf776297d/test/app/scriptureforge/activity/activity.e2e-spec.ts#L512)):*
<details><summary><i>Previously, irahopkinson (Ira Hopkinson) wrote…</i></summary><blockquote>

> ```
> element(by.className('activity-username'))
> ```
Can this be moved back into the activity page object? We want to hide this behind a named function so when it changes we only need to change it in the page object file. Especially since it is repeated below.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io/reviews/sillsdev/web-languageforge/255#-:-LC_bzB04xB-2UX8iSL4:b-c29t90)*
<!-- Sent from Reviewable.io -->

irahopkinson commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

The first two LanguageData tests fail only on a developer machine reminding us that we still need to fix them (they are ignored in the build system).

To fix the third, search in `deploy/dependencies.yml` for `geo`. I guess those changes were merged in after you had already done a deploy. I would manually fix it up (rather than running Ansible). You'll need to install the package, change the config file, then run the `geoipupdate` to get the database.

---

Reviewed 7 of 7 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[src/Api/Model/Scriptureforge/Sfchecks/Command/TextCommands.php, line 54 at r1](https://reviewable.io/reviews/sillsdev/web-languageforge/255#-LCWtpLa3lQMvCPw0nbp:-LC_hUULBbp0PTJSSFHG:b-80slcl) ([raw file](https://github.com/sillsdev/web-languageforge/blob/078c6266a8acc3bebfc8dbcf3cac0c7bf776297d/src/Api/Model/Scriptureforge/Sfchecks/Command/TextCommands.php#L54)):*
<details><summary><i>Previously, nigel-wells (Nigel Wells) wrote…</i></summary><blockquote>

Done - have updated IDE settings to stop this happening again as well.
</blockquote></details>

Sweet

---


*Comments from [Reviewable](https://reviewable.io/reviews/sillsdev/web-languageforge/255#-:-LC_iVHa1JbIX68D998H:b-mg600k)*
<!-- Sent from Reviewable.io -->
